crates/common/download/src/download.rs

```diff
@@ -85,21 +85,6 @@ impl DownloadInfo {
     }
 }
 
-/// Possible authentication schemes
-#[derive(Debug, Clone, Deserialize, PartialEq, Eq, Serialize)]
-#[serde(rename_all = "camelCase")]
-#[serde(deny_unknown_fields)]
-pub enum Auth {
-    /// HTTP Bearer authentication
-    Bearer(String),
-}
```

Comment:
Have you consider to keep this type, i.e. to add a `Basic` case?

Using strings for auth headers is convenient but also error prone. This is why I would not throw away this type too quickly. The main issue I see is related to the current crate organization we have: we don't want to have all HTTP related crate depending on this download crate.

Reply:
> Have you consider to keep this type, i.e. to add a Basic case?

Yes, but there are two mixed feelings. Then I decided to discard it.
- One side, in the `C8YJwtRetriever` level, keep returning `Result<String, JwtError>` will make less changes in existing code. 
- Another side, in the `downloader` level, since it's using `reqwest` directly (on the contrary `c8y_http_proxy` is using a wrapper of `hyper`), this `Auth` enum helps because `reqwest` has a dedicated API to add auth like `request.bearer_auth(token)`.

> The main issue I see is related to the current crate organization we have: we don't want to have all HTTP related crate depending on this download crate.

Exactly. Another issue. Then question is where is appropriate... maybe create a new file at `crates/common/tedge_utils/http.rs`? 

If we keep `enum`,  the return type of `C8YJwtRetriever` and `C8YBasicAuthProvider` to `Result<Auth, JwtError>` can be changed. Then implement functions on `Auth` to return string with `Bearer` or `Basic` prefix.

```rust
pub enum Auth {
    /// HTTP Bearer authentication
    Bearer(String),
    Basic(String),
}

impl Auth {
    pub fn to_header_value(&self) -> String {
        match self {
            Auth::Bearer(value) => format!("Bearer {value}"),
            Auth::Basic(value) => format("Basic {value}"),
        }
    }
}
```

But again, mixed feeling for efforts vs beauty. :(

However, I'm not sure how this `enum` will be helpful compared to the changes that is required due to the return type change. The point is, we use different HTTP crates for downloader (`reqwest`) and http_proxy_api (`hyper`).

Reply:
> But again, mixed feeling for efforts vs usability. :(

My concern is not related to usability but maintainability. Using `String` values is convenient until you have to refactor your code: you have then no support from the compiler for which a bearer string or a basic-auth string make no difference.

> Probably the best way is that `C8YBasicAuthProvider` returns `Result<(HeaderName, HeaderValue), Error)`. See my [comment](https://github.com/thin-edge/thin-edge.io/pull/3190#issuecomment-2419148275).

Why not. This makes the intent clearer.

Reply:
The use of `HeaderMap` is far better than trying to keep this adhoc `Auth` type.